sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -328,6 +329,35 @@ export class ApplicationStudentsController extends BaseController {
     }
   }
 
+  /**
+   * Cancels an in-progress change request for an existing student application.
+   * @param applicationId application ID of the change request to be cancelled.
+   */
+  @Patch(":applicationId/cancel-change-request")
+  @ApiNotFoundResponse({
+    description: "Not able to find the in-progress change request.",
+  })
+  async applicationCancelChangeRequest(
+    @Param("applicationId", ParseIntPipe) applicationId: number,
+    @UserToken() studentToken: StudentUserToken,
+  ): Promise<void> {
+    try {
+      await this.applicationService.cancelApplicationChangeRequest(
```

Comment:
Are we planning to lock the change request application on `ministry approval` part to avoid both approval, and cancellation happening at same time?

Reply:
Do you mean a lock to ensure that between the moment of selecting the record and checking to the expected status, and the moment of executing the DB update, there is a concurrent change?

Reply:
I will take a note to review it once we do the Ministry approval ticket.